spec.emu

```diff
@@ -676,134 +765,210 @@
       </table>
     </emu-table>
 
+    <emu-clause id="sec-mappings-grammar">
+      <h1>Mappings grammar</h1>
+      <p>The mappings String must adhere to the following grammar:</p>
+
+      <emu-grammar type="definition">
+        MappingsField :
+          SegmentList
+
+        SegmentList :
+          Segment
+          Segment `;` SegmentList
+
+        Segment :
+          MappingList?
+
+        MappingList :
+          Mapping
+          Mapping `,` MappingList
+
+        Mapping :
+          GeneratedColumn
+          GeneratedColumn OriginalSource OriginalLine OriginalColumn Name?
+
+        GeneratedColumn :
+          Vlq
+
+        OriginalSource :
+          Vlq
+
+        OriginalLine :
+          Vlq
+
+        OriginalColumn :
+          Vlq
+
+        Name :
+          Vlq
+      </emu-grammar>
+
+      <p>A <dfn id="decode-mapping-state-record" variants="Decode Mapping State Records">Decoded Mapping State Record</dfn> has the following fields:</p>
+      <emu-table id="table-decode-mapping-state-fields" caption="Fields of Decode Mapping State Records">
+        <table>
+          <thead>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value Type
+              </th>
+            </tr>
+          </thead>
+          <tr>
+            <td>[[GeneratedLine]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+          <tr>
+            <td>[[GeneratedColumn]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+          <tr>
+            <td>[[SourceIndex]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+          <tr>
+            <td>[[OriginalLine]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+          <tr>
+            <td>[[OriginalColumn]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+          <tr>
+            <td>[[NameIndex]]</td>
+            <td>a non-negative integer</td>
+          </tr>
+        </table>
+      </emu-table>
+
+      <emu-clause id="sec-DecodeMappingsField" type="sdo">
+        <h1>
+          DecodeMappingsField (
+            _state_: a Decode Mapping State Record,
```

Comment:
Very minor: this doesn't get hyperlinked because the variant specified above is plural but this isn't

Reply:
Thanks! The &lt;dfn&gt; had a typo.